### PR TITLE
Automatic ingestion from S3 sample

### PR DESCRIPTION
**Pull request by avdvir**

### CODE REVIEW

File: S3/S3EventGridPoc/Readme.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# Automatic Ingestion from S3 to ADx using lambda
+## Deploy the lambda
+Find instructions [here]( https://docs.aws.amazon.com/lambda/latest/dg/gettingstarted-package.html)
+## Define AAD application
+* Find instruction [here]( https://docs.microsoft.com/en-us/azure/active-directory/develop/howto-create-service-principal-portal)
```
Comment (ohadbitt) [inline S3/S3EventGridPoc/Readme.md]:
```suggestion
* Find instruction [here]( https://docs.microsoft.com/azure/active-directory/develop/howto-create-service-principal-portal)
```

File: S3/S3EventGridPoc/Readme.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+* Define a key for the application and save it.
+* Grant the application permissions on your ADx cluster - at least Ingestor for all the databases you want to ingest to.
```
Comment (ohadbitt) [inline S3/S3EventGridPoc/Readme.md]:
```suggestion
* Grant the application permissions on your ADX cluster - at least database Ingestor on the database you want to ingest to.
```

File: S3/S3EventGridPoc/Readme.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+## Define environment variables
+* Under Lambda -> Configuration -> Environment variables define the following
+* AppKey - your aad app key
+*	AppId - your aad app Id
```
Comment (ohadbitt) [inline S3/S3EventGridPoc/Readme.md]: Capitalize all of these: AAD, AWS, ADX
```suggestion
*	AppId - your AAD app Id
```

File: S3/S3EventGridPoc/Function.cs [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+
+namespace S3EventGridPoc;
+
+public class Function
+{
+    IAmazonS3 S3Client { get; set; }
+
+    /// <summary>
+    /// Default constructor. This constructor is used by Lambda to construct the instance. When invoked in a Lambda environment
+    /// the AWS credentials will come from the IAM role associated with the function and the AWS region will be set to the
```
Comment (ohadbitt) [inline S3/S3EventGridPoc/Function.cs:20]: You say the credentials will come from IAM but you ask for AwsCredentials parameter ... which one is it ?

File: S3/S3EventGridPoc/Function.cs [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+                FlushImmediately = true
+            };
+
+            var sourceOptions = new StorageSourceOptions() { DeleteSourceOnSuccess = false, Size = s3.Object.Size };
+            var uri = $"https://{s3.Bucket.Name}.s3.{record.AwsRegion}.amazonaws.com/{s3.Object.Key}";
+            Console.WriteLine($"start to ingest {uri}");
+            await client.IngestFromStorageAsync(uri:$"{uri};AwsCredentials={awsCredentials}", ingestionProperties: kustoIngestionProperties, sourceOptions);
+            Console.WriteLine($"complete to ingest {uri}");
+
+            client.Dispose();
```
Comment (ohadbitt) [inline S3/S3EventGridPoc/Function.cs]: dispose outside the loop

File: S3/S3EventGridPoc/aws-lambda-tools-defaults.json [hunk 6]
```diff
@@ -0,0 +1,15 @@
+{
+  "Information": [
```
Comment (ohadbitt) [inline S3/S3EventGridPoc/aws-lambda-tools-defaults.json:2]: Maybe move or copy this info to the README ? 
People could miss this file

**Review by ohadbitt — APPROVED** [review]

File: S3/S3EventGridPoc/aws-lambda-tools-defaults.json (hunk 6 above)
Comment (avdvir) [reply S3/S3EventGridPoc/aws-lambda-tools-defaults.json:2]: This is auto generated by lambda, nothing that I wrote

File: S3/S3EventGridPoc/Function.cs (hunk 4 above)
Comment (avdvir) [reply S3/S3EventGridPoc/Function.cs:20]: this is also part of the auto generated code. actually I don't need this client

**Pull request merged**